mama/c_cpp/src/c/subscription.c

```diff
@@ -451,20 +443,32 @@ mamaSubscription_setupBasic (
         return MAMA_STATUS_NO_BRIDGE_IMPL;
     }
 
-#ifdef WITH_ENTITLEMENTS
+
     mamaBridgeImpl* bridge = mamaSubscription_getBridgeImpl(subscription);
-    if (gEntitlementClient == 0 || mamaBridgeImpl_areEntitlementsDeferred(bridge))
+    if (0 == mamaInternal_getEntitlementBridgeCount() || mamaBridgeImpl_areEntitlementsDeferred(bridge))
     {
         mama_log (MAMA_LOG_LEVEL_FINER,
                     "Entitlements checking at subscription creation deferred to %s bridge [%p]",
                     bridge->bridgeGetName(), bridge);
     }
     else
-        self->mSubjectContext.mOeaSubscription = oeaClient_newSubscription (&entitlementStatus, gEntitlementClient);
-#endif
+    {
+        mamaEntitlementBridge mamaEntBridge = NULL;
+        mama_status status = mamaTransportImpl_getEntitlementBridge(transport, &mamaEntBridge);
 
-    /*Up from entitlement check based on string compare on symbol*/
-    if (!isEntitledToSymbol (source, symbol, self))
+        if (NULL != mamaEntBridge)
+        {
+            mamaEntBridge->createSubscription(mamaEntBridge, &self->mSubjectContext);
+        }
+        else
+        {
+            mama_log(MAMA_LOG_LEVEL_ERROR,
+                     "mamaSubscription_setupBasic(): Could not find entitlement bridge!");
+            return MAMA_STATUS_NO_BRIDGE_IMPL;
+        }
+    }
+
+    if (!isEntitledToSymbol (source, symbol, self, transport))
```

Comment:
this can be moved to AFTER the subject context is populated with the transport to allow the function interface change above.